### DISCUSSION TITLE
Ollama support

### DISCUSSION THREAD

**Discussion by dpark12300** [music-assistant/support#2309] [Feature requests and ideas]
We already have OpenAI support, so why not make it even more self-hosted, and use Ollama instead? HASS already has integration with Ollama, so Music Assistant should too.

**Comment by OzGav** [comment]
Have you tried? I don’t know anything about Ollama but if you can change the conversation agent then maybe it will work?

**Comment by janstadt** [comment]
I have an instance of ollama stood up and have a few configurations going, but i cant figure out how to make a holistic conversation agent that can both manage HA stuff (turning things on/off, etc) as well as controlling MASS. I used this integration: https://github.com/acon96/home-llm as well as https://github.com/jekalmin/extended_openai_conversation and both work fine for one or the other, but not both. I wish there was a single setup/tutorial that showed me how to use a voice assistant to control both MASS and HASS. Still futzing around with it though so hopefully i'll figure something out.

**Reply by OzGav** [reply]
How is all this going since the 2024.6.0 HA update ?

**Reply by OzGav** [reply]
How is this going since the 2024.8.0 HA update?

**Reply by janstadt** [reply]
I havent been using voice lately for playing stuff but ill try this weekend to see if anything has improved.

**Reply by janstadt** [reply]
Direct integration with ollama for me isnt going super great for music purposes but the Home LLM integration pointing to my ollama instance is working pretty well. I think it has to do with how its been optimized to use their specific model for controlling hass/mass.

**Reply by markselias** [reply]
How did you set this up? Would you mind sharing which model, prompt and additional setup you did in order to get this running? I can't get it to work properly.

**Reply by janstadt** [reply]
The way i had it working is no longer supported. You need to configure some blueprints in HA now for it to work: https://github.com/music-assistant/voice-support
